src/Template/UpdateChecker.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace App\Template;
+
+use Symfony\Component\HttpClient\CachingHttpClient;
+use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Component\HttpKernel\HttpCache\Store;
+use Symfony\Component\HttpKernel\KernelInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+class UpdateChecker
+{
+    private HttpClientInterface $client;
+
+    public function __construct(
+        KernelInterface $kernel,
+        ?HttpClientInterface $httpClient = null
+    ) {
+        if ('test' === $kernel->getEnvironment()) {
+            if (null === $httpClient) {
+                throw new \InvalidArgumentException('An explicit HttpClient must be provided during testing.');
+            }
+            $this->client = $httpClient;
+        } else {
+            $this->client = new CachingHttpClient(
+                HttpClient::create(),
+                new Store("{$kernel->getCacheDir()}/releases")
+            );
+        }
```

Comment:
This code now behaves differently in a test environment. To a certain extent, that's desirable: we want to get an exception in a testing environment if no explicit httpClient is provided. However, I'd argue that this different behaviour should be limited to notices. Therefore my suggestion would be to restructure this to
```php
        if (null === $httpClient) {
            if ('test' === $kernel->getEnvironment()) {
                throw new \InvalidArgumentException('An explicit HttpClient must be provided during testing.');
            }
            $this->client = $httpClient;
        } else {
            $this->client = new CachingHttpClient(
                HttpClient::create(),
                new Store("{$kernel->getCacheDir()}/releases")
            );
        }
        ```

Reply:
You're right, I see now how my suggestion is flawed. However, I feel the logic to change things still stands. Therefore, I propose the following:
```php
        if (null === $httpClient) {
            if ('test' === $kernel->getEnvironment()) {
                throw new \InvalidArgumentException('An explicit HttpClient must be provided during testing.');
            }
            $this->client = new CachingHttpClient(
                HttpClient::create(),
                new Store("{$kernel->getCacheDir()}/releases")
            );
        } else {
            $this->client = $httpClient;
        }
```
With this structure, it should behave similar to your code, with the notable exception of calling the constructor with a HttpClient in any other environment than test; with the current code it would be ignored, while this structure always allows for a prepopulated value.